common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -576,6 +661,7 @@ public MergeStatuses finalizeShuffleMerge(FinalizeShuffleMerge msg) {
           } finally {
             partition.closeAllFilesAndDeleteIfNeeded(false);
           }
+          cleanUpAppShufflePartitionInfoInDB(partition.appAttemptShuffleMergeId);
```

Comment:
Why are we doing this ? We just inserted it in the `compute` above ?
We should have a test which validates that after successful finalization, the db contains the partition info.

Reply:
Test to be added

Reply:
Removed the cleanup though.

Reply:
Just found that we do have a UT which tests that a [finalized shuffle](https://github.com/apache/spark/pull/35906/files#diff-6dac15558c856c231c284e221b8ae0cdd2ad79c98586eb7e49c3ddcefa53a241R263), after a restart, getOrCreateAppShufflePartitionInfo will trigger [BlockPushNonFatalFailure](https://github.com/apache/spark/pull/35906/files#diff-6dac15558c856c231c284e221b8ae0cdd2ad79c98586eb7e49c3ddcefa53a241R289) with an exception message containing "is finalized". This UT should have tested that the DB contains the partitionInfo after a successful finalization.

Reply:
If we had a test for this, curious why it was not surfacing this issue as a test failure. Perhaps something needs to be augmented in the test ? Or some other bug ?

Reply:
I just tried locally to revert this change, the UT doesn't fail. Will double check why it is not failing if we have write first and then remove immediately here, and update. Reopen the issue.

Reply:
Figured out the issue. It is not a bug within the RemoteBlockPushResolver, but the UT was not triggering this part of the code path. It is missing one shuffle merge partition preparation after a service restart. Since non-finalized shuffle merge will not be written into DB, during restart, there will be empty entry for the ShuffleMergedPartitions. In order to trigger this code path, other than directly calling finalizeShuffleMerge, we should call the prepareAppShufflePartition again, as shown in [this commit](https://github.com/apache/spark/pull/35906/commits/7b697e3166854d0df8fd8701baef5de4cde64fc1), and then call the finalizeShuffleMerge. The UT will walk through this part of the code. I enhanced the UT as shown in the commit, and if I added the cleanup DB operation, same as shown in this comment, the current UT will fail. Thanks for catching this issue.